### PR TITLE
feat: Apply recommended security best practices

### PR DESCRIPTION
**Pull request by project0**
### What does this PR do?

* All deployments have strong default security context configured
* Ensure all deployments follow best practice with shared test

### Motivation

Increase trust of usage and better security defaults.

### More

- [x] Yes, I updated the tests accordingly
- [x] Yes, I ran `make test` and all the tests passed

### CODE REVIEW

**Review by mloiseleur — APPROVED** [review]
LVGTM

File: hub-agent/tests/__snapshot__/.empty
Comment (youkoulayley) [inline hub-agent/tests/__snapshot__/.empty:1]: Why do we need to commit this directory?

**Review by youkoulayley — CHANGES_REQUESTED** [review]

File: hub-agent/tests/__snapshot__/.empty
Comment (mloiseleur) [reply hub-agent/tests/__snapshot__/.empty:1]: Last version of helm-unittest needs a folder __ __snapshot__ __.

Comment (tomMoulard) [reply hub-agent/tests/__snapshot__/.empty:1]: If it is just for testing purposes, what do you think about creating the folder when executing tests, and removing it from git ?

The makefile could look something like:

```Makefile
__snapshot__:
	mkdir __snapshot__

test: __snapshot__
	docker run ${DOCKER_ARGS} --entrypoint /bin/sh --rm -v $(CURDIR):/charts -w /charts helmunittest/helm-unittest:3.11.1-0.3.0 /charts/hack/test.sh
```

Comment (mloiseleur) [reply hub-agent/tests/__snapshot__/.empty:1]: the alternative would be to a mount a tmpfs, see https://github.com/traefik/hub-helm-chart/pull/80/commits/d4ea7afbcc67f8ed3f4310e2a99d21050890e66c

Comment (mloiseleur) [reply hub-agent/tests/__snapshot__/.empty:1]: @tomMoulard: Interesting idea. I will give it a try.

Comment (mloiseleur) [reply hub-agent/tests/__snapshot__/.empty:1]: It works. Thanks @tomMoulard ! Are we good, @youkoulayley ?

**Review by youkoulayley — APPROVED** [review]

**Pull request merged**